### PR TITLE
[EVM] Support PRINTF instruction

### PR DESCRIPTION
**Pull request by shermike**
* PRINTF instruction is supported as a builtin with variable arguments.
* Support Global Init function. This function calls constructors for all
  global objects. It is invoked before the target function.
* Several improvements in EVM Asm parser:
  - support sections switching: .data and .text directives
  - support .size and .type directives
  - support data objects

### CODE REVIEW

**Comment by shermike** [comment]
cc @avm

File: llvm/lib/MC/EVMBinaryObjectWriter.cpp [hunk 1]
```diff
@@ -128,6 +128,10 @@ uint64_t EVMBinaryObjectWriter::writeObject(MCAssembler &Asm,
     Json.attribute("size", Size);
     Json.attribute("visible", Visible);
 
+    if (StringRef(FuncName).startswith("_GLOBAL__sub_I_")) {
```
Comment (avm) [inline llvm/lib/MC/EVMBinaryObjectWriter.cpp:131]: What does this prefix mean? A comment would be helpful (but a helper function with a meaningful name might be even better).

File: llvm/projects/evm-sdk/evm-stdlib/stdlib.s [hunk 2]
```diff
@@ -226,6 +232,20 @@ modpow_cont:
     # Stack: [result, return_address]
     JUMP
 
+# This function registers a function(typically destructor) to be called by exit
```
Comment (avm) [inline llvm/projects/evm-sdk/evm-stdlib/stdlib.s:235]: Space missing before `(`.

File: llvm/projects/evm-sdk/evm-stdlib/stdlib.s [hunk 3]
```diff
@@ -226,6 +232,20 @@ modpow_cont:
...
+# Arguments:
+# - address of the function to be called
+# - argument that will be passed to the function
+# - dso_handle
+# Does nothing for a now.
```
Comment (avm) [inline llvm/projects/evm-sdk/evm-stdlib/stdlib.s:240]: "for now"

File: llvm/projects/evm-sdk/evm-stdlib/stdlib.s [hunk 4]
```diff
@@ -226,6 +232,20 @@ modpow_cont:
...
+STDLIB_FUNC __cxa_atexit
```
Comment (avm) [inline llvm/projects/evm-sdk/evm-stdlib/stdlib.s:241]: Why don't we implement `__cxa_atexit` in C?

File: llvm/lib/MC/EVMBinaryObjectWriter.cpp (hunk 1 above)
Comment (shermike) [reply llvm/lib/MC/EVMBinaryObjectWriter.cpp:131]: This prefix is used by LLVM for global init function. Will add some comment https://github.com/NilFoundation/zkllvm-circifier/blob/48fd30c40daef638c21db6f7dd33e4d3f97438c9/clang/lib/CodeGen/CGDeclCXX.cpp#L897

File: llvm/projects/evm-sdk/evm-stdlib/stdlib.s (hunk 4 above)
Comment (shermike) [reply llvm/projects/evm-sdk/evm-stdlib/stdlib.s:241]: Since it does nothing at the moment, the assembler version is simpler and more optimal.

**Review by makxenov — APPROVED** [review]

**Pull request merged**